vernac/vernacentries.ml

```diff
@@ -1458,7 +1458,10 @@ let vernac_reserve bl =
     let env = Global.env() in
     let sigma = Evd.from_env env in
     let t,ctx = Constrintern.interp_type env sigma c in
-    let t = Detyping.detype Detyping.Now false Id.Set.empty env (Evd.from_ctx ctx) t in
+    let t = Flags.without_option Detyping.print_universes (fun () ->
+        Detyping.detype Detyping.Now false Id.Set.empty env (Evd.from_ctx ctx) t)
```

Comment:
It's quite confusing that something called `print_universe` affects the behaviour of pretyping...

Reply:
Is detyping not part of printing?

Reply:
Pretyping, not detyping. The root of the problem is that pattern interpretation goes through detyping.